Add safeguards to prevent javascript error

billyhunt opened the pull request:
[Link to ZenHub issue.](https://github.com/usdoj-crt/crt-portal-management/issues/1030)

## What does this change?

The close button on the trafficking modal is not working.  There is an error that happens, shown below.  This PR fixes the error by not scrolling when the form does not allow it.  

## Screenshots (for front-end PR):

![image](https://user-images.githubusercontent.com/6232068/130081963-a59ea83d-e3b9-4e21-b303-f151331067bf.png)
![image](https://user-images.githubusercontent.com/6232068/130082158-d51a82bd-69ed-40a9-9527-9a43a5e97bc5.png)


## Checklist:

### Author

+ [x] If this is a story, run locally and check to make sure all Acceptance Criteria are met. (If any criteria are unclear, ask about them.).
+ [x] Check for, document, and establish a testing plan for any behavior that may vary across environments or is otherwise difficult to test.
+ [x] Check for [accessibility](/docs/a11y_plan.md).
+ [x] [Tests pass](https://github.com/USDOJ/crt-portal/#tests).

### Reviewer

+ [ ] If this is a story, run locally and check to make sure all Acceptance Criteria are met. (If any criteria are unclear, ask about them.).
+ [ ] Check for any behavior that may vary across environments or is difficult to test, and ensure that it is well-understood, documented, and that there is a testing plan in place.
+ [ ] Re-check for [accessibility](/docs/a11y_plan.md).
+ [ ] [Tests pass](https://github.com/USDOJ/crt-portal/#tests).

## Notes for reviewer:

See [PR instructions doc](https://github.com/usdoj/crt-portal/blob/master/docs/pull_requests.md) for full pull request review instructions.

louh commented:
So this does fix the issue, however, I wonder if it's worth considering correcting the logic in the function arguments itself.

On [line 58 of `modal.js`](https://github.com/usdoj-crt/crt-portal/blob/develop/crt_portal/static/js/modal.js#L58):

```js
 root.CRT.cancelModal = function(modal_el, cancel_el, form_el = {}) {
```

When `form_el` is not passed to the `cancelModal()` method (which is [the case here](https://github.com/usdoj-crt/crt-portal/blob/develop/crt_portal/static/js/redirect-modal.js#L34)) it is assigned a default value of empty object.

The original [`if` statement](https://github.com/usdoj-crt/crt-portal/blob/develop/crt_portal/static/js/modal.js#L60) checks to see if `form_el` is a falsy value:

```js
if (form_el) {
  // Do stuff
}
```

However, an empty object is _not_ falsy. This is why the `if` statement is always true, and this block will always run.

Rather than check to see if the object is not empty (and has the methods it wants to run) I believe we might be able to fix the problem by not assigning a default value to `form_el` at all. If `form_el` is undefined, it will pass the falsy check and it should avoid running the `if` block.

The benefit of doing this is some cleaner code, and the conditions on the `if` statement won't need to be updated with the behavior inside its block in the future.

billyhunt commented:
> So this does fix the issue, however, I wonder if it's worth considering correcting the logic in the function arguments itself.
> 
> On [line 58 of `modal.js`](https://github.com/usdoj-crt/crt-portal/blob/develop/crt_portal/static/js/modal.js#L58):
> 
> ```js
>  root.CRT.cancelModal = function(modal_el, cancel_el, form_el = {}) {
> ```
> 
> When `form_el` is not passed to the `cancelModal()` method (which is [the case here](https://github.com/usdoj-crt/crt-portal/blob/develop/crt_portal/static/js/redirect-modal.js#L34)) it is assigned a default value of empty object.
> 
> The original [`if` statement](https://github.com/usdoj-crt/crt-portal/blob/develop/crt_portal/static/js/modal.js#L60) checks to see if `form_el` is a falsy value:
> 
> ```js
> if (form_el) {
>   // Do stuff
> }
> ```
> 
> However, an empty object is _not_ falsy. This is why the `if` statement is always true, and this block will always run.
> 
> Rather than check to see if the object is not empty (and has the methods it wants to run) I believe we might be able to fix the problem by not assigning a default value to `form_el` at all. If `form_el` is undefined, it will pass the falsy check and it should avoid running the `if` block.
> 
> The benefit of doing this is some cleaner code, and the conditions on the `if` statement won't need to be updated with the behavior inside its block in the future.

That makes a lot of sense.  I will give it a go.

billyhunt commented:
I updated the implementation to @louh's  suggestion, tested all the modals, and fixed the unsupported browser modal, which had a different bug from last year. (good idea to check them all @ilodidoj)

louh reviewed (approved):
LGTM!

ilodidoj reviewed (commented):
@billyhunt I'll be merging this. Please also create one PR for the Release 08-24-2021 merge.
